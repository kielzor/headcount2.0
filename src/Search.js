import React, { Component } from 'react'

class Search extends Component {
  render () {
    return (
      <div className='input-box'>
        <h1 className='header-text'>Headcount 2.0</h1>
        <input 
          className='search-input' 
          type='text' 
          placeholder='Filter districts' 
          onChange={(e) => this.props.updateCards(e.target.value)}
        />
      </div>
    )
  }
}

export default Search