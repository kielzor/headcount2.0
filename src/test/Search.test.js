import React from 'react'
import Search from '../Search.js'
import { shallow, mount } from 'enzyme'

describe('Search', () => {
  let updateCardsMock;
  let wrapper;

  beforeEach(() => {
    updateCardsMock = jest.fn()
    wrapper = shallow(<Search updateCards={updateCardsMock} />)
  })

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('invokes updateCards when input is changed', () => {
    const mockEvent =  {target: {value: 'Col'}}

    wrapper.find('.search-input').simulate('change', mockEvent)

    expect(updateCardsMock).toHaveBeenCalled()
  })

  it('returns the correct value from the input field', () => {
    const mockEvent = {target: {value: 'COL'}}

    wrapper.find('.search-input').simulate('change', mockEvent)

    expect(mockEvent.target.value).toBe('COL')
  })
})