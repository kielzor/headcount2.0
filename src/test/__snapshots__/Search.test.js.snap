// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search
    updateCards={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="header-text"
>
          Headcount 2.0
</h1>,
        <input
          className="search-input"
          onChange={[Function]}
          placeholder="Filter districts"
          type="text"
/>,
      ],
      "className": "input-box",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Headcount 2.0",
          "className": "header-text",
        },
        "ref": null,
        "rendered": "Headcount 2.0",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "search-input",
          "onChange": [Function],
          "placeholder": "Filter districts",
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="header-text"
>
            Headcount 2.0
</h1>,
          <input
            className="search-input"
            onChange={[Function]}
            placeholder="Filter districts"
            type="text"
/>,
        ],
        "className": "input-box",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Headcount 2.0",
            "className": "header-text",
          },
          "ref": null,
          "rendered": "Headcount 2.0",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "search-input",
            "onChange": [Function],
            "placeholder": "Filter districts",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
